### ISSUE TITLE
[ceremonies] refactor monolithic files

### ISSUE THREAD

**Issue by clangenb** [encointer/pallets#257]
The ceremonies pallet has been growing constantly over the past years and is now very unreadable, I propose a refactoring like:

* **lib.rs**: only contains the pallet's macro stuff. Every dispatchable shall map to exactly one module method.
* **module_impls/**: implementation on the module for the extrinsics or other helpers, aka `impl Module<T> {..}` blocks
   * **registration.rs**: contains module implementations for registering, updating_regstration, unregistering, endorsing
   * **meetup.rs**: contains module implementations for attest_claims, and claim_rewards
   * **common.rs**: contains other module implementations 
* **storage_helpers.rs**: contains generic code to handle the bootstrapper, reputable, etc. storage maps.
* **test:** refactor tests to multiple files. I think it should be one file per extrinsic complex extrinsic, and a system test file, and maybe another one that combines a few simple extrinsics.
   * register_participant.rs
   * ... other per extrinsic files
   * **storage_setters.rs:** `set_location_tolerance` etc.
   * **system_tests.rs**: contains community growth tests etc.

**Comment by brenzi** [comment]
I don't think we want a folder module_impls. Modules can stay in src/, right?

**Comment by clangenb** [comment]
Yeah it is not necessary. I just thought I might be nice to have this hierarchical organization, not sure if it really is nice though.
